components/server/src/websocket/websocket-connection-manager.ts

```diff
@@ -376,31 +377,31 @@ class GitpodJsonRpcProxyFactory<T extends object> extends JsonRpcProxyFactory<T>
     protected async onRequest(method: string, ...args: any[]): Promise<any> {
         const span = TraceContext.startSpan(method, undefined);
         const userId = this.clientMetadata.userId;
-        const requestId = span.context().toTraceId();
-        return runWithLogContext(
-            "request",
+        const abortController = new AbortController();
+        const cancellationToken = args[args.length - 1];
+        if (CancellationToken.is(cancellationToken)) {
+            cancellationToken.onCancellationRequested(() => abortController.abort());
+        }
+        return runWithRequestContext(
             {
-                userId,
-                contextId: requestId,
-                method,
+                requestKind: "jsonrpc",
+                requestMethod: method,
+                signal: abortController.signal,
+                subjectId: userId ? SubjectId.fromUserId(userId) : undefined,
+                traceId: span.context().toTraceId(),
             },
             () => {
                 try {
                     // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-                    return this.internalOnRequest(span, requestId, method, ...args);
+                    return this.internalOnRequest(span, method, ...args);
```

Comment:
should not we await here? otherwise span in finally does not make sense?

Reply:
Unrelated, but totally correct! Will add.

Reply:
Done.